src/lib/modules.nix

```diff
@@ -0,0 +1,222 @@
+{
+  config,
+  dlib,
+  lib,
+}: let
+  l = lib // builtins;
+
+  configFuncMsg = ''
+    consider passing a path to a file instead of a function
+    - functions can't be encoded to JSON, and as such most features of
+    dream2nix won't work because of this since they require passing around
+    the config as JSON.
```

Comment:
They would still work in situations where everything is pure so that's why I allowed it, but if you think that's not needed then we can just change the `warn` to an `abort`.

Reply:
I think we should not allow the user to do things that potentially break other features. Therefore better not allow it in the first place.

Reply:
Done in ae9a29002f11db00e4dd9a60dc938f120ceb41ea